lib/types.ts

```diff
@@ -1,10 +1,12 @@
 import { IncomingMessage, ServerResponse } from 'http';
 
-export type SourceListDirective = false | (string | ((req: IncomingMessage, res: ServerResponse) => string))[];
-export type PluginTypesDirective = false | (string | ((req: IncomingMessage, res: ServerResponse) => string))[];
-export type SandboxDirective = false | (string | ((req: IncomingMessage, res: ServerResponse) => string))[];
-export type ReportUriDirective = false | (string | ((req: IncomingMessage, res: ServerResponse) => string));
-export type RequireSriForDirective = false | (string | ((req: IncomingMessage, res: ServerResponse) => string))[];
+export type DirectiveType = (string | ((req: IncomingMessage, res: ServerResponse) => string));
```

Comment:
Does this type need to be exported? If not, could we remove the `export`?

Also, could this instead be an `interface`?

Reply:
Of course, it was automatically, i always export everything hahahah, let me fix it

Reply:
n to com fome ainda amor,

Reply:
Also this cannot be an interface because it has to be an object in order to be an interface. And this is not the case

Reply:
Done!

Reply:
Is that true? What about [this section of the TypeScript documentation](https://www.typescriptlang.org/docs/handbook/interfaces.html#function-types)?

Reply:
Sorry! that was new to me!

Thank you! I've added it!

Reply:
i actually ended up not understanding how to add in the case that we have... hence it triggered an error... what would be the correct way in this case, were we can also have a string without a function call?

Reply:
Ah, I think you're right—a `type` is probably correct.